templates/rag-conversation-zep/rag_conversation_zep/chain.py

```diff
@@ -0,0 +1,149 @@
+import os
+from operator import itemgetter
+from typing import List, Tuple
+
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain.prompts.prompt import PromptTemplate
+from langchain.schema import AIMessage, HumanMessage, format_document
+from langchain.schema.output_parser import StrOutputParser
+from langchain.schema.runnable import (
+    RunnableBranch,
+    RunnableLambda,
+    RunnableMap,
+    RunnablePassthrough,
+)
+from langchain.vectorstores.zep import CollectionConfig, ZepVectorStore
+from pydantic import BaseModel, Field
+
+ZEP_API_URL = os.environ.get("ZEP_API_URL", "http://localhost:8000")
+ZEP_API_KEY = os.environ.get("ZEP_API_KEY", None)
+ZEP_COLLECTION_NAME = os.environ.get("ZEP_COLLECTION", "langchaintest")
+
+# Zep offers native, hardware-accelerated MMR. Enabling this will improve
+# the diversity of results, but may also reduce relevance. You can tune
+# the lambda parameter to control the tradeoff between relevance and diversity.
+# Enabling is a good default.
+ZEP_MMR = True
+ZEP_MMR_LAMBDA = 0.5
+
+collection_config = CollectionConfig(
+    name=ZEP_COLLECTION_NAME,
+    description="Zep collection for LangChain",
+    metadata={},
+    embedding_dimensions=1536,
+    is_auto_embedded=True,
+)
+
+
+# ## Ingest code - you may need to run this the first time
+# # Load
+# from langchain.document_loaders import WebBaseLoader
+
+# loader = WebBaseLoader("https://lilianweng.github.io/posts/2023-06-23-agent/")
+# data = loader.load()
+
+# # Split
+# from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+# text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=0)
+# all_splits = text_splitter.split_documents(data)
+
+# # Add to vectorDB
+# from langchain.embeddings import FakeEmbeddings
+
+# vectorstore = ZepVectorStore.from_documents(
+#     documents=all_splits,
+#     collection_name=ZEP_COLLECTION_NAME,
+#     config=collection_config,
+#     api_url=ZEP_API_URL,
+#     api_key=ZEP_API_KEY,
+#     embedding=FakeEmbeddings(size=1),
+# )
+# retriever = vectorstore.as_retriever()
+
+
+vectorstore = ZepVectorStore(
+    collection_name=ZEP_COLLECTION_NAME,
+    config=collection_config,
+    api_url=ZEP_API_URL,
+    api_key=ZEP_API_KEY,
+    embedding=None,
+)
+retriever = vectorstore.as_retriever(
+    search_type="mmr" if ZEP_MMR else "similarity",
+    search_kwargs={"k": 4, "lambda_mult": ZEP_MMR_LAMBDA},
```

Comment:
would be cool to make search_type and lambda_mult[ configurable](https://python.langchain.com/docs/expression_language/how_to/configure)

```python
from langchain.schema.runnable import ConfigurableField

retriever = vectorstore.as_retriever().configurable_fields(
  search_type = ConfigurableField(
    id="search_type", name="Search type"
  ),
  search_kwargs = ConfigurableField(
    id="search_kwargs", name="Search kwargs"
  ),
)
```